File: MATLAB/Toolbox/SDK/Fluigent/enums/fgt_ERROR_CODE.m
classdef fgt_ERROR_CODE < uint8
    %FGT_ERROR_CODE Error states reported by the SDK
    
    enumeration
        OK (0),
        USB_error (1),
        Wrong_command (2),
        No_module_at_index (3),
        Wrong_module (4),
        Module_is_sleep (5),
        Master_error (6),
        Failed_init_all_instr (7),
        Wrong_parameter (8),
        Overpressure (9),
        Underpressure (10),
        No_instr_found (11),
        No_modules_found (12),
        No_pressure_controller_found (13),
        Calibrating (14),
        Dll_dependency_error (15),
        Processing (16)
    end
    
    properties
    end
    
    methods
    end
    
end

File: C#/fgt_sdk_csharp/Enums/fgt_ERROR_CODE.cs
namespace fgt_sdk.Enums
{
    /// <summary>
    /// Returned error codes from dll functions
    /// </summary>
    public enum fgt_ERROR_CODE
    {
        /// No error
        OK,
        /// USB communication error
        USB_error,
        /// Wrong command was sent
        Wrong_command,
        /// There is no module initialized at selected index
        No_module_at_index,
        /// Wrong module was selected, unavailable feature
        Wrong_module,
        /// Module is in sleep mode, orders are not taken into account
        Module_is_sleep,
        /// Controller error
        Master_error,
        /// Some instruments failed to initialize
        Failed_init_all_instr,
        /// Function parameter is not correct or out of the bounds
        Wrong_parameter,
        /// Pressure module is in overpressure protection
        Overpressure,
        /// Pressure module is in underpressure protection
        Underpressure,
        /// No Fluigent instrument was found
        No_instr_found,
        /// No Fluigent pressure or sensor module was found
        No_modules_found,
        /// No Fluigent pressure controller was found
        No_pressure_controller_found,
        /// Pressure or sensor module is calibrating, read value may be incorrect
        Calibrating,
        /// Some dependencies are not found
        Dll_dependency_error,
        /// M-Switch is still processing the last input
        Processing,
    }
}